src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -292,5 +290,24 @@ private void EndSendFileInternal(IAsyncResult asyncResult)
             }
 
         }
+
+        private static readonly bool IsWindows7OrBefore;
+
+        static Socket()
```

Comment:
How about putting this into a static method that's then used to initialize the field rather than using a cctor? e.g.
```C#
private static readonly bool s_isWindows7OrBefore = IsWindows7OrBefore();
private static bool IsWindows7OrBefore()
{
    Version osVersion = Environment.OSVersion.Version;
    return osVersion.Major < 6 || (osVersion.Major == 6 && osVersion.Minor <= 1);
}
```
Unless something's changed, having an explicit cctor prevents usage of beforefieldinit on the type.